exporters/prometheus/src/exporter_utils.cc

```diff
@@ -313,7 +381,7 @@ void PrometheusExporterUtils::SetMetricBasic(
   // Note that attribute keys are sorted, but sanitized keys can be out-of-order.
   // We could sort the sanitized keys again, but this seems too expensive to do
   // in this hot code path. Instead, we ignore out-of-order keys and emit a warning.
-  metric.label.reserve(labels.size() + 2);
+  metric.label.reserve(labels.size());
```

Comment:
Why this ?

Space is still needed to add a scope_name and scope_version.

Reply:
Ah, I probably caused some confusion... The previous iteration of this PR added `job` and `instance` labels, which meant I asked that +2 to be removed in https://github.com/open-telemetry/opentelemetry-cpp/pull/2301#discussion_r1342680813.  But now, after rebasing and adding scope name and scope verison, the +2 needs to stay.

Reply:
Ok, makes sense. Resolved then, Owent just fixed it.